OpenSim/Examples/DataTable/example.cpp

```diff
@@ -0,0 +1,357 @@
+  SimTK::RowVector_<SimTK::Vec6> row_vec6{static_cast<int>(data1_vec6.size()), 
+                                          data1_vec6.data()};
+  dt_real.addRow(row_real);
+  dt_vec3.addRow(row_vec3);
+  dt_vec6.addRow(row_vec6);
+
+  // Add a single row to the DataTable using an iterator. Note that this 
+  // function is different from the one used above, which is addRows().
+  dt_real.addRow(data1_real.cbegin(), data1_real.cend());
+  dt_vec3.addRow(data1_vec3.cbegin(), data1_vec3.cend());
+  dt_vec6.addRow(data1_vec6.cbegin(), data1_vec6.cend());
+
+  // Trying to add a row using an iterator that does not produce enough 
+  // elements to fill up the row will throw as follows.
+  try {
+    dt_real.addRow(data1_real.cbegin(), data1_real.cend() - 1);
+  } catch(OpenSim::InvalidEntry&) {}
+  try {
+    dt_vec3.addRow(data1_vec3.cbegin(), data1_vec3.cend() - 1);
+  } catch(OpenSim::InvalidEntry&) {}
+  try {
+    dt_vec6.addRow(data1_vec6.cbegin(), data1_vec6.cend() - 1);
+  } catch(OpenSim::InvalidEntry&) {}
+
+  // DataTable can be told to allow for missing values when using iterator
+  // as follows. In this case, the missing values will be SimTK::NaN. See the
+  // documentation for addrow for details on the arguments. The third argument
+  // is ignored in this call.
+  dt_real.addRow(data1_real.cbegin(), data1_real.cend() - 1, 0, true);
+  dt_vec3.addRow(data1_vec3.cbegin(), data1_vec3.cend() - 1, 0, true);
+  dt_vec6.addRow(data1_vec6.cbegin(), data1_vec6.cend() - 1, 0, true);
+
+  // DataTable has the same interface for adding columns as well.
+  // Add a column using a SimTK::Vector_.
+  std::vector<SimTK::Real> data2_real(7, 1);
+  std::vector<SimTK::Vec3> data2_vec3(7, {1, 2, 3});
+  std::vector<SimTK::Vec6> data2_vec6(7, {1, 2, 3, 4, 5, 6});
+  SimTK::Vector_<SimTK::Real> col_real{static_cast<int>(data2_real.size()),
+                                       data2_real.data()};
+  SimTK::Vector_<SimTK::Vec3> col_vec3{static_cast<int>(data2_vec3.size()),
+                                       data2_vec3.data()};
+  SimTK::Vector_<SimTK::Vec6> col_vec6{static_cast<int>(data2_vec6.size()),
+                                       data2_vec6.data()};
+  dt_real.addCol(col_real);
+  dt_vec3.addCol(col_vec3);
+  dt_vec6.addCol(col_vec6);
+
+  // Add a column using an iterator.
+  dt_real.addCol(data2_real.cbegin(), data2_real.cend());
+  dt_vec3.addCol(data2_vec3.cbegin(), data2_vec3.cend());
+  dt_vec6.addCol(data2_vec6.cbegin(), data2_vec6.cend());
+
+  // Add multiple columns using an iterator.
+  for(int i = 0; i < 7; ++i) {
+    data2_real.push_back(i);
+    data2_vec3.push_back(SimTK::Vec3{1, 2, 3} + i);
+    data2_vec6.push_back(SimTK::Vec6{1, 2, 3, 4, 5, 6} + i);
+  }
+  dt_real.addCols(data2_real.cbegin(), data2_real.cend());
+  dt_vec3.addCols(data2_vec3.cbegin(), data2_vec3.cend());
+  dt_vec6.addCols(data2_vec6.cbegin(), data2_vec6.cend());
+
+  // Copy construct DataTable.
+  decltype(dt_real) dt_real_copy{dt_real};
```

Comment:
I wanted to make copy construction explicit because I am demonstrating copy construction here.
```
auto dt_real_copy{dt_real}
```
will not work because the type of `dt_real_copy` will be deduced to be `std::initializer_list<OpenSim::DataTable_<double>>` which is not what I want.
```
auto dt_real_copy = dt_real;
```
will work. At the time of writing I thought it was better. Now I think auto will work equally fine. Will change it.